### PR TITLE
Aik hierarchy

### PR DESCRIPTION
**Pull request by DenisKarch**
Move AIK to Storage hierarchy.

### CODE REVIEW

File: attest/tpm_linux.go [hunk 1]
```diff
@@ -190,7 +190,13 @@ func (t *TPM) getPrimaryKeyHandle(pHnd tpmutil.Handle) (tpmutil.Handle, bool, er
 		return pHnd, false, nil
 	}
 
-	ekHnd, _, err := tpm2.CreatePrimary(t.rwc, tpm2.HandleEndorsement, tpm2.PCRSelection{}, "", "", defaultEKTemplate)
+	var ekHnd tpmutil.Handle
+	switch pHnd {
+	case commonSrkEquivalentHandle:
+		ekHnd, _, err = tpm2.CreatePrimary(t.rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", "", defaultSRKTemplate)
```
Comment (twitchy-jsonp) [inline attest/tpm_linux.go]: Please revise the name to be more generic.

File: attest/tpm_linux.go [hunk 2]
```diff
@@ -393,7 +407,7 @@ func (t *TPM) MintAIK(opts *MintOptions) (*Key, error) {
 		}()
 
 		// We can only certify the creation immediately afterwards, so we cache the result.
-		attestation, sig, err := tpm2.CertifyCreation(t.rwc, "", keyHandle, keyHandle, nil, creationHash, tpm2.SigScheme{tpm2.AlgRSASSA, tpm2.AlgSHA256, 0}, tix)
+		attestation, sig, err := tpm2.CertifyCreation(t.rwc, "", keyHandle, keyHandle, nil, creationHash, tpm2.SigScheme{tpm2.AlgRSASSA, tpm2.AlgSHA256, 0}, &tix)
```
Comment (twitchy-jsonp) [inline attest/tpm_linux.go]: Does it work without taking the pointer to `tix`?

File: attest/tpm_linux.go [hunk 3]
```diff
@@ -408,6 +422,7 @@ func (t *TPM) MintAIK(opts *MintOptions) (*Key, error) {
 			KeyEncoding:       KeyEncodingParameterized,
```
Comment (twitchy-jsonp) [inline attest/tpm_linux.go]: Should this be a different `KeyEncoding`?

Comment (twitchy-jsonp) [reply attest/tpm_linux.go]: (such as `KeyEncodingEncrypted`)

File: attest/tpm_linux.go (hunk 1 above)
Comment (twitchy-jsonp) [reply attest/tpm_linux.go]: (`ekHnd`)

File: attest/tpm_linux.go (hunk 2 above)
Comment (DenisKarch) [reply attest/tpm_linux.go]: nope, would not build with just tix. (CreateKeyEx returned a *Ticket, but CreateKey returns Ticket)

File: attest/tpm_linux.go (hunk 3 above)
Comment (DenisKarch) [reply attest/tpm_linux.go]: Probably yes, should we also drop the check for KeyEncoding when loading?

File: attest/tpm_linux.go (hunk 2 above)
Comment (twitchy-jsonp) [reply attest/tpm_linux.go]: Got it, cheers.

File: attest/tpm_linux.go (hunk 3 above)
Comment (twitchy-jsonp) [reply attest/tpm_linux.go]: The intent of the `KeyEncoding` field is to describe how the serialized key is represented. `KeyEncodingParameterized` should indicate to `LoadKey` that the key is a primary key that needs to be generated to get the key handle. `KeyEncodingEncrypted` symbolizes that the encrypted private blob is present, and the key needs to be loaded to get the handle.

The aim of this was to support keys that were generated with an older version of the library.

So either:

1. We should change the check of KeyEncoding to `KeyEncodingEncrypted`, dropping support for loading the old key format
2. We should use a switch to support loading both.

My preference is the second, but its up to you.

**Comment by DenisKarch** [comment]
don't pull this in yet btw. waiting for go-tpm to approve a change for CreateKey https://github.com/google/go-tpm/pull/85, will update the go.mod when that is sorted.

File: attest/tpm_linux.go (hunk 3 above)
Comment (DenisKarch) [reply attest/tpm_linux.go]: I've already made the change for the 1st option

**Comment by twitchy-jsonp** [comment]
Sounds good, lets merge this when the dependencies are ready.

**Pull request merged**